docs/developer_guide/savu_hebi.rst

```diff
@@ -0,0 +1,48 @@
+Savu and hebi integration overview
+==================================
+
+Savu_ is a package developed at Diamond for processing tomographic data which is
+integrated into mantidimaging via Hebi_, a RESTful API. The aim of the integration
+is to be able to process data with savu's processing plugins via the gui, with a
+similar UX to using the tomopy filters built in to the 'filters' window. Savu is
+designed to run in a variety of environments, which makes the integration somewhat
+complicated.
+
+Savu workflow
+-------------
+
+Savu works by applying the plugins specified in a process list to the data in a folder.
+
+A process list:
+ - Is a nexus (.nxs) file
+ - Starts with a loader plugin
+ - Has one or more processing plugins
+ - Optionally ends with a saver plugin
+ - Internally to savu, can be built with `savu_config`
+
+Hebi
+----
+
+Hebi_ is a REST api on top of savu which creates and monitors 'jobs' running in savu.
+It is WIP, and the API can be expected to evolve over time.
+
+Imaging to hebi to savu and back
+--------------------------------
+
+How everything is connected:
+ - |container_setup|
+ - An http client is connected to hebi to submit jobs, ask for plugin details etc.
+ - A websocket client waits for messages from hebi, signalling when jobs complete.
+
+To apply a 'savu filter' to an image stack:
+ - |process_list_built|
+ - Submits a job to hebi with the pl and path to the input data.
```

Comment:
```suggestion
 - Submits a job to hebi with the process list and path to the input data.
```

Reply:
There is one more step after this. After the job is submitted, the client checks the HTTP Response code. If it's 200 (HTTP OK) then it [joins](https://github.com/mantidproject/mantidimaging/blob/4deeaa847d5120a57b5fcec26df63986d0fbd814/mantidimaging/gui/windows/savu_filters/sio_model.py#L30) a WebSocket [room](https://socket.io/docs/rooms-and-namespaces/). The output from the job is sent to that "room", and clients in it get notified when something is broadcast into it.